omero_marshal/encode/encoders/shape.py

```diff
@@ -33,7 +34,41 @@ def encode(self, obj):
         self.set_if_not_none(v, 'TheC', obj.theC)
         self.set_if_not_none(v, 'TheT', obj.theT)
         self.set_if_not_none(v, 'TheZ', obj.theZ)
+        self.set_if_not_none(
+            v, 'Transform', self.encode_transform(obj.transform))
         self.set_if_not_none(v, 'Visible', obj.visibility)
         return v
 
+    @staticmethod
+    def encode_transform(transform):
+        if not transform:
+            return
+        if isinstance(transform, RType):
+            transform = transform.getValue()
+        if transform == 'none':
+            return
+
+        tr, args = transform[:-1].split('(')
+        a = map(float, args.split(' '))
+
+        if tr == 'matrix':
+            pass
+        elif tr == 'translate':
+            a = [1.0, 0.0, 0.0, 1.0, a[0], a[1] if len(a) > 1 else 0.0]
+        elif tr == 'scale':
+            a = [a[0], 0.0, 0.0, a[-1], 0.0, 0.0]
+        else:
+            raise ValueError('Unknown transformation "%s"' % transform)
+
+        return {
+            '@type': 'TBD#AffineTransform',
```

Comment:
The type is `http://www.openmicroscopy.org/Schemas/ROI/2015-01#AffineTransform`.